File: JavaPractice/src/FindPlayerScoreHighestLowest.java
import java.util.HashMap;

/**
In a given game there are 10 players(can grow more). Each individual will have there own score. What we are interested is
1) Get score of any given player
2) Get total score of all players
3) Get which player scored lowest
4) Get which player scored highest
5) Show the players scored from highest to lowest
6) Show the players scored from lowest to highest
**/
public class FindPlayerScoreHighestLowest {

	public static class Players {
		
		private String name = null;
		private int minscore = Integer.MAX_VALUE;
		private int maxscore = Integer.MIN_VALUE;
		
		public Players(String string, int i) {
			setName(string);
			setScore(string,i);
		}
		public String getName() {
			return name;
		}
		public void setName(String name) {
			this.name = name;
		}
		public int getMinscore() {
			return minscore;
		}
		public void setMinscore(int minscore) {
			this.minscore = minscore;
		}
		public int getMaxscore() {
			return maxscore;
		}
		public void setMaxscore(int maxscore) {
			this.maxscore = maxscore;
		}

		public void setScore(String name, int score) {
			
			setName(name);
			if(score < getMinscore()) {
				setMinscore(score);
			}
			if(score > getMaxscore()) {
				setMaxscore(score);
			}
		}
		
	}
	public static void main(String[] args) {
		// TODO Auto-generated method stub
		HashMap<String, Players> hm = new HashMap<>();
		FindPlayerScoreHighestLowest.Players pl = new FindPlayerScoreHighestLowest.Players("Divyesh", 5);
		if(!hm.containsKey("Divyesh"))
			hm.put("Divyesh", pl);
		System.out.println(hm.get("Divyesh").getMaxscore());
		System.out.println(hm.get("Divyesh").getMinscore());
		
		
	}
}

File: PythonPractice/src/FindPlayerScoreHighestLowest.py
class Players:
    def __init__(self, name: str, score: int):
        if not name or score is None:
            raise ValueError("Invalid input: Name and score must be provided")
        self.name = name
        self.minscore = score
        self.maxscore = score

    def get_name(self) -> str:
        return self.name

    def get_minscore(self) -> int:
        return self.minscore

    def set_minscore(self, minscore: int):
        self.minscore = minscore

    def get_maxscore(self) -> int:
        return self.maxscore

    def set_maxscore(self, maxscore: int):
        self.maxscore = maxscore

    def set_score(self, name: str, score: int):
        if not name or score is None:
            raise ValueError("Invalid input: Name and score must be provided")
        self.name = name
        if score < self.get_minscore():
            self.set_minscore(score)
        if score > self.get_maxscore():
            self.set_maxscore(score)

def main():
    hm = {}
    pl = Players("Divyesh", 5)

    if "Divyesh" not in hm:
        hm["Divyesh"] = pl

    print(hm["Divyesh"].get_maxscore())  # Get max score of player "Divyesh"
    print(hm["Divyesh"].get_minscore())  # Get min score of player "Divyesh"

if __name__ == "__main__":
    main()
